apm-agent-plugins/apm-dubbo-plugin/src/main/java/co/elastic/apm/agent/dubbo/advice/ApacheDubboFilterAdvice.java

```diff
@@ -0,0 +1,145 @@
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.dubbo.advice;
+
+import co.elastic.apm.agent.bci.HelperClassManager;
+import co.elastic.apm.agent.bci.VisibleForAdvice;
+import co.elastic.apm.agent.dubbo.helper.ApacheDubboAttachmentHelper;
+import co.elastic.apm.agent.dubbo.helper.DubboApiInfo;
+import co.elastic.apm.agent.dubbo.helper.DubboTraceHelper;
+import co.elastic.apm.agent.dubbo.helper.IgnoreExceptionHelper;
+import co.elastic.apm.agent.impl.ElasticApmTracer;
+import co.elastic.apm.agent.impl.Scope;
+import co.elastic.apm.agent.impl.transaction.Span;
+import co.elastic.apm.agent.impl.transaction.Transaction;
+import net.bytebuddy.asm.Advice;
+import org.apache.dubbo.rpc.Invocation;
+import org.apache.dubbo.rpc.Invoker;
+import org.apache.dubbo.rpc.Result;
+import org.apache.dubbo.rpc.RpcContext;
+
+@VisibleForAdvice
+public class ApacheDubboFilterAdvice {
+
+    @VisibleForAdvice
+    public static ElasticApmTracer tracer;
+
+    @VisibleForAdvice
+    public static HelperClassManager<ApacheDubboAttachmentHelper> helperClassManager;
+
+    public static void init(ElasticApmTracer tracer) {
+        ApacheDubboFilterAdvice.tracer = tracer;
+        DubboTraceHelper.init(tracer);
+        IgnoreExceptionHelper.init(tracer);
+        helperClassManager = HelperClassManager.ForAnyClassLoader.of(tracer,
+            "co.elastic.apm.agent.dubbo.helper.ApacheDubboAttachmentHelperImpl");
+    }
+
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static void onEnterFilterInvoke(@Advice.Argument(1) Invocation invocation,
+                                           @Advice.Local("span") Span span,
+                                           @Advice.Local("apiClazz") Class<?> apiClazz,
+                                           @Advice.Local("transaction") Transaction transaction,
+                                           @Advice.Local("scope") Scope scope) {
+        RpcContext context = RpcContext.getContext();
+        String version = context.getUrl().getParameter("version");
+        Invoker<?> invoker = invocation.getInvoker();
+        apiClazz = invoker.getInterface();
+        DubboApiInfo dubboApiInfo = new DubboApiInfo(
+            apiClazz, invocation.getMethodName(),
+            invocation.getParameterTypes(), version);
```

Comment:
Rather than allocating an object, just pass all the information as method parameters to the helper

Reply:
good for performance, ok